Add path element to returned properties for nodes

wabson opened the pull request:
Note this is an optional property, only present when include=paths is specified on
request. However without this property definition this optional information is not visible via a generated API client.

montgolfiere commented:
Many thanks, Will. LGTM so I will pull this in. 

Also relates to https://issues.alfresco.com/jira/browse/REPO-489 more generally.

montgolfiere commented:
Will ... sorry, I see that you just deleted your branch. However, we may need to revert this, so that we look at it again. 

1. Typo in NodeMinimal - should be PathInfo not PathElement
2. SwaggerHub gives a warning (we don't have any others)

Swagger Warning
Extra JSON Reference properties will be ignored: required

montgolfiere commented:
The "build" checks also picked up on at least #1 (not sure about #2)

https://bamboo.alfresco.com/bamboo/browse/DEV-RAE9-3

Reverting the PR, fixed the "build" 

https://bamboo.alfresco.com/bamboo/browse/DEV-RAE9-4

wabson commented:
Thanks @montgolfiere for spotting that. I'm not sure how it worked in my testing with the bad reference (item 1) but I've fixed that now.

Regarding the `required` flag (item 2) I mis-understood this. Rather than being declared at the property level it is actually an array of required property names, which should be declared on the schema item alongside `properties`. See http://swagger.io/specification/#schemaObject, this approach is used in `NodeMinimalEntry` thus to declare `entry` as a required property.

````
  NodeMinimalEntry:
    type: object
    required:
      - entry
    properties:
      entry:
        $ref: '#/definitions/NodeMinimal'
```

In contrast `NodeMinimal` does not specify a `required` list, so by default all properties on it are optional.

I've restored the branch that I deleted and fixed both issues, if this sounds OK then I will open a new PR for us to continue the conversation there.

montgolfiere commented:
Yes, please open a new PR, thanks. 

Please note: we also have an outstanding JIRA issue to fix other missing / optional properties in the future (REPO-489) although we will do that initially on our "develop" branch. If you need any others then please let us know, so that we can prioritise that sooner.
